src/components/Calculator.js

```diff
@@ -1,103 +1,96 @@
+import React, { useState } from 'react';
 import '../styles/Calculator.css';
 import calculate from '../logic/calculate';
 
-export default class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      outputStr: '0',
-      nextString: false,
-      total: null,
-      next: null,
-      operation: null,
-    };
-  }
+export default function Calculator() {
+  const [outputStr, setOutPutString] = useState('0');
+  const [nextString, SetNextString] = useState(false);
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
 
-dropValue = (e) => {
-  const X = e.target.value.toString();
+  const dropValue = (e) => {
+    const X = e.target.value.toString();
 
-  if (X === '-' || X === '+' || X === '÷' || X === 'x' || X === '%') {
-    this.setState({ nextString: !this.state.nextString });
-    this.setState({ operation: e.target.value });
-    this.setState({ outputStr: '0' });
-  } else if ((X.match(/^[0-9]+$/) !== null && this.state.nextString === false) || X === '.') {
-    if (this.state.total === null) {
-      this.setState({ total: X });
-      this.setState({ outputStr: X });
-    } else {
-      this.setState({ total: `${this.state.total}${e.target.value}` });
-      this.setState({ outputStr: this.state.total + X });
-    }
-  } else if ((X.match(/^[0-9]+$/) !== null && this.state.nextString === true) || X === '.') {
-    if (this.state.next === null) {
-      this.setState({ next: X });
-      this.setState({ outputStr: X });
-    } else {
-      this.setState({ next: `${this.state.next}${e.target.value}` });
-      this.setState({ outputStr: this.state.next + X });
-    }
-  } else if (X === 'AC') {
-    this.setState({ outputStr: '0' });
-    this.setState({ total: null });
-    this.setState({ next: null });
-    this.setState({ operation: null });
-    this.setState({ nextString: false });
-  } else if (X === '+/-') {
-    if (this.state.next !== null) {
-      this.setState({ next: `${-1 * parseInt(this.state.next, 10)}` });
-      this.setState({ outputStr: `${-1 * parseInt(this.state.outputStr, 10)}` });
-    } else {
-      this.setState({ total: `${-1 * parseInt(this.state.total, 10)}` });
-      this.setState({ outputStr: `${-1 * parseInt(this.state.outputStr, 10)}` });
+    if (X === '-' || X === '+' || X === '÷' || X === 'x' || X === '%') {
+      SetNextString(!nextString);
+      setOperation(X);
+      setOutPutString('0');
+    } else if ((X.match(/^[0-9]+$/) !== null && nextString === false) || X === '.') {
+      if (total === null) {
+        setTotal(X);
+        setOutPutString(X);
+      } else {
+        setTotal(`${total}${e.target.value}`);
+        setOutPutString(total + X);
+      }
+    } else if ((X.match(/^[0-9]+$/) !== null && nextString === true) || X === '.') {
+      if (next === null) {
+        setNext(X);
+        setOutPutString(X);
```

Comment:
- [ ] Great job however after factoring the Calculator component from a class-based component to a functional component using [hooks](https://reactjs.org/docs/hooks-intro.html). Everything must still work as expected. I record a short video here: [link](https://www.loom.com/share/189a64d99d544d7491cc9f782bdff12c) showing how you calculate work, Kindly fix that. 😄